controller/src/main/java/io/pravega/controller/fault/ControllerClusterListener.java

```diff
@@ -126,24 +120,79 @@ protected void startUp() throws InterruptedException, Exception {
                 }
             };
 
-            taskSweeper.sweepOrphanedTasks(processes);
+            sweepEventProcessorReaders(processes);
+            sweepTransactions(processes);
+            sweepTasks(processes);
+
+            log.info("Controller cluster listener startUp complete");
+        } finally {
+            LoggerHelpers.traceLeave(log, objectId, "startUp", traceId);
+        }
+    }
+
+    private void handleHostRemoved(Host host) {
+        taskSweeper.sweepOrphanedTasks(host.getHostId());
+
+        eventProcessorsOpt.ifPresent(controllerEventProcessors -> {
```

Comment:
This was optional from before, and i agree with you that in production we will always have event processors system initialized so this is probably only for testing. Will remove optional as part of a separate issue though.. created issue